setup.py

```diff
@@ -1,48 +1,70 @@
 #! /usr/bin/env python
 
 import os
+import io 
 import sys
 import tempfile
 import shutil
 import builtins
 import textwrap
 
 import setuptools
+from configparser import ConfigParser
 from distutils.command.build_py import build_py
 from distutils.command.sdist import sdist
 from distutils.errors import CompileError, LinkError
 
+# Checking if Python installation is >=3.6
+if sys.version_info < (3, 6, 0):
 
-DISTNAME = 'scikit-image'
-DESCRIPTION = 'Image processing in Python'
-MAINTAINER = 'Stefan van der Walt'
-MAINTAINER_EMAIL = 'stefan@sun.ac.za'
-URL = 'https://scikit-image.org'
-LICENSE = 'Modified BSD'
-DOWNLOAD_URL = 'https://scikit-image.org/docs/stable/install.html'
-PROJECT_URLS = {
-    "Bug Tracker": 'https://github.com/scikit-image/scikit-image/issues',
-    "Documentation": 'https://scikit-image.org/docs/stable/',
-    "Source Code": 'https://github.com/scikit-image/scikit-image'
-}
-
-with open('README.md', encoding='utf-8') as f:
-    LONG_DESCRIPTION = f.read()
-
-if sys.version_info < (3, 6):
-
-    error = """Python {py} detected.
-
+    error = f"""Python {'.'.join([str(v) for v in sys.version_info[:3]])}
```

Comment:
you can't have fstrings before python 3.6. So when this error is hit, it need to be in syntax that is compatible with old versions of python.

Namely, we are still interested in providing this error message to those that are still using python 2.7 for whatever reason.

I think it is best to revert.